src/backend/distributed/connection/placement_connection.c

```diff
@@ -483,13 +490,35 @@ StartPlacementListConnection(uint32 flags, List *placementAccessList,
 		ShardPlacementAccess *placementAccess =
 			(ShardPlacementAccess *) linitial(placementAccessList);
 		ShardPlacement *placement = placementAccess->placement;
+		char *nodeName = placement->nodeName;
+		int nodePort = placement->nodePort;
 
 		/*
 		 * No suitable connection in the placement->connection mapping, get one from
 		 * the node->connection pool.
 		 */
-		chosenConnection = StartNodeConnection(flags, placement->nodeName,
-											   placement->nodePort);
+		chosenConnection = StartNodeConnection(flags, nodeName, nodePort);
```

Comment:
Moved the main connection finding logic into a separate `FindPlacementListConnection` function.